docs/conf.py

```diff
@@ -74,3 +76,57 @@
 intersphinx_mapping = {
     'numpy': ('https://numpy.org/doc/stable/', None),
 }
+
+
+#  -- Automatic generation of API reference pages -----------------------------
+
+autosummary_generate = True
+remove_from_toctrees = ["reference/*"]
+
+
+def rstjinja(app, docname, source):
+    """
+    Render our pages as a jinja template for fancy templating goodness.
+    """
+    # https://www.ericholscher.com/blog/2016/jul/25/integrating-jinja-rst-sphinx/
+    # Make sure we're outputting HTML
+    if app.builder.format != "html":
+        return
+    src = source[0]
+    rendered = app.builder.templates.render_string(src, app.config.html_context)
+    source[0] = rendered
+
+
+def get_module_functions(module, exclude=None):
+    """Return a list of function names for the given submodule."""
+    
+    mod = getattr(shapely, module)
+    return mod.__all__
+
+
+html_context = {
+    'get_module_functions': get_module_functions
+}
+
+
+template = """
+.. autogenerated file
+
+.. currentmodule:: shapely
+
+.. autosummary::
+   :toctree: reference/
+
+"""
+
+modules = ["_geometry", "creation", "constructive", "coordinates", "io", "linear", "measurement", "predicates", "set_operations"]
+functions = [func for mod in modules for func in get_module_functions(mod)]
+
+template += "   " + "\n   ".join(functions)
+
+with open("_reference.rst", "w") as f:
+    f.write(template)
```

Comment:
The reason for creating this otherwise ignored `_reference.rst` file is because the autogeneration of the reference/ stub files by `autosummary` happens _before_ updates the source with jinja rendering (and so at that point, the `.. autosummary::` directives do not yet contain the methods, and otherwise the individual pages would not be autogenerated)

Reply:
Suggest adding a brief comment to the code to note this.